tests/unit/s2n_security_rules_test.c

```diff
@@ -0,0 +1,409 @@
+                EXPECT_TRUE(result.found_error);
+
+                size_t output_size = s2n_stuffer_data_available(&result.output);
+                EXPECT_EQUAL(output_size, strlen(expected_output));
+                uint8_t *output_bytes = s2n_stuffer_raw_read(&result.output, output_size);
+                EXPECT_NOT_NULL(output_bytes);
+                EXPECT_BYTEARRAY_EQUAL(expected_output, output_bytes, output_size);
+            };
+
+            /* Writes correct name for versioned policy */
+            {
+                const char expected_prefix[] =
+                        "Test Rule: policy test_all: ";
+                const size_t expected_prefix_size = strlen(expected_prefix);
+
+                DEFER_CLEANUP(struct s2n_security_rule_result result = { 0 },
+                        s2n_security_rule_result_free);
+                EXPECT_SUCCESS(s2n_stuffer_growable_alloc(&result.output, 100));
+
+                EXPECT_OK(s2n_security_rule_validate_policy(
+                        &test_rule, &security_policy_test_all, &result));
+                EXPECT_TRUE(result.found_error);
+
+                size_t output_size = s2n_stuffer_data_available(&result.output);
+                EXPECT_TRUE(output_size > expected_prefix_size);
+                uint8_t *output_bytes = s2n_stuffer_raw_read(&result.output, expected_prefix_size);
+                EXPECT_NOT_NULL(output_bytes);
+                EXPECT_BYTEARRAY_EQUAL(output_bytes, expected_prefix, expected_prefix_size);
+            };
+
+            /* Test: Truncates output if insufficient memory for output */
+            {
+                const char full_output[] =
+                        "Test Rule: policy unnamed: curve: secp384r1 (#1)\n";
+                const size_t full_output_size = strlen(full_output);
+                struct {
+                    uint32_t mem_size;
+                    const char expected_output[sizeof(full_output)];
+                } test_cases[] = {
+                    /* No truncation if sufficient space */
+                    {
+                            .mem_size = full_output_size,
+                            .expected_output = "Test Rule: policy unnamed: curve: secp384r1 (#1)\n" },
+                    /* Truncate if necessary */
+                    {
+                            .mem_size = full_output_size - 1,
+                            .expected_output = "Test Rule: policy unnamed: curve: secp384r1 ...\n" },
+                    /* Print only the truncation message */
+                    {
+                            .mem_size = strlen("...\n"),
+                            .expected_output = "...\n" },
+                    /* Truncate the truncation message */
+                    {
+                            .mem_size = 3,
+                            .expected_output = "..\n" },
+                    { .mem_size = 1,
+                            .expected_output = "\n" },
```

Comment:
nit: The formatting is a bit weird here. Or does clang format force it this way?

Reply:
Oh gross, I missed that clang-format did that -_-

That is what clang-format does, but I remember that there's a way to work around it.